### ISSUE TITLE
Pytest-sugar errors out when running with deselected test.

### ISSUE THREAD

**Issue by aselus-hub** [Frozenball/pytest-sugar#87]
py.test running in python3.4 on ubuntu 14.04 with a command that deselects at least one test execution fails with a `AttributeError: 'Function' object has no attribute 'when'` exception after the test exeution block(NOTE: at least one test hasto be actually 'deselected', it does not matter if any run). 

Running with '-p no:sugar' does fix the issue. 

example bellow:
 
     $python3 -m pytest -m noomarker test_py.py
```
Test session starts (platform: linux, Python 3.4.0, pytest 2.9.1, pytest-sugar 0.7.0)
rootdir: /home/alexei, inifile: 
plugins: cpp-0.4, capturelog-0.7, sugar-0.7.0, timeout-1.0.0

================================================================================= 1 tests deselected by "-m 'noomarker'" =================================================================================

Results (0.01s):
Traceback (most recent call last):
  File "/usr/lib/python3.4/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.4/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.4/dist-packages/pytest.py", line 17, in <module>
    raise SystemExit(pytest.main())
  File "/usr/local/lib/python3.4/dist-packages/_pytest/config.py", line 49, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 596, in execute
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/main.py", line 119, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/main.py", line 114, in wrap_session
    exitstatus=session.exitstatus)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
    wrap_controller.send(call_outcome)
  File "/usr/local/lib/python3.4/dist-packages/_pytest/terminal.py", line 378, in pytest_sessionfinish
    self.summary_stats()
  File "/usr/local/lib/python3.4/dist-packages/pytest_sugar.py", line 436, in summary_stats
    if self.count('deselected') > 0:
  File "/usr/local/lib/python3.4/dist-packages/pytest_sugar.py", line 397, in count
    [x for x in self.stats.get(key) if x.when in when]
  File "/usr/local/lib/python3.4/dist-packages/pytest_sugar.py", line 397, in <listcomp>
    [x for x in self.stats.get(key) if x.when in when]
AttributeError: 'Function' object has no attribute 'when'
```

the code for the example test:
```
class TestClass:
  def test_nothing(self):
    pass
```

**Comment by Frozenball** [comment]
Thanks for the bug report.

**Comment by Frozenball** [comment]
Fixed in 0.7.1.
